content/2020-06-19-how-to-learn-typescript-quickly/index.md

```diff
@@ -0,0 +1,259 @@
+
+In this case with TypeScript, maybe you say, "I'm going to write one line of TypeScript a day." Sounds trivial, right? Good, it should. You start there, establish the habit, then build upwards and outwards. Upwards meaning you continue doing it every day. Outwards meaning you increase the lines of code (without losing the habit).
+
+### Share with Others
+
+> Pass on what you have learned, Luke.
+>
+> —Yoda
+
+One of the most important parts of staying on track is sharing with others and [Learning in Public](https://www.swyx.io/writing/learn-in-public/). There are various mediums you can choose from for sharing:
+
+- Tweets
+- Blog Posts
+- Podcasts
+- Videos
+- Tutorials
+- Courses
+
+Any of these ways work - and there are probably more I didn't list. This will aid your project for a few reasons:
+
+- Help you retain what you're learning
+- Build expertise in the area
+- Meet others who are learning
+
+When I was learning TypeScript, I tweeted as I learned. Not only did I meet other TypeScript community members, but people corrected me or taught me new things. Towards the end, people started tweeted me asking for TypeScript help. I didn't always know the answer, but my sharing with others and learning in public helped me build **reputation** and **credibility**.
+
+I even ended up attracting the attention of an editor for a tech blog who paid me to write a few TypeScript articles. As you can see, it pays to share with others.
+
+### Keep Your Focus
+
+The last piece I want to touch on for staying on track is keeping your focus. To stay on track, you have to keep yourself focused. The first and foremost is **limit your focus**. Ideally, learning TypeScript is your main focus. I wouldn't suggest doing this and learning Rust or another language or something else. Keeping your attention on one project increases the likelihood you'll stay on track.
+
+The other thing you might want to try is [**deep work**](https://www.calnewport.com/books/deep-work/). An idea by Cal Newport, it's where you create an environment that lets you focus on that project for a certain amount of time (such as 1 hour or more) without distractions. Figure out when you're most productive. For me, it's in the morning before work. I wake up early, go through my routine, and then spend some time learning.
+
+Lastly, find tools to help you stay focused. I am a big fan of the **Pomodoro Technique**. I have an app called [Stretchly](https://hovancik.net/stretchly/) that reminds me to take a 20-second break every 20 minutes and a 5-minute break every 40 minutes. This allows me to focus for that period and then rest. It then repeats itself like a cycle.
+
+## Find the Flag
+
+The last and final leg of this learning journey is finding the flag. You want to make it to the finish line and feel accomplished when you get there. Here, you want to use systems that will help you practice, build confidence, remember vocabulary, and go for depth so that you find the checkered flag.
```

Comment:
```suggestion
The last and final leg of this learning journey is finding the flag. You want to make it to the finish line and feel accomplished when you get there. Here, you want to use systems that will help you practice, build confidence, remember vocabulary, and go for depth so that you find the checkered flag.
```
```suggestion
The last and final leg of this learning journey is finding the flag. You should make it to the finish line and feel accomplished when you get there. Here, you want to use systems that will help you practice, build confidence, remember vocabulary, and go for depth. This will lead to the beloved checkered flag.
```